fluss-lake/fluss-lake-paimon/src/test/java/com/alibaba/fluss/lake/paimon/lakehouse/PaimonSortedRecordReaderTest.java

```diff
@@ -0,0 +1,602 @@
+        for (int i = 0; i < 10; i++) {
+            GenericRow row =
+                    isPartitioned
+                            ? row(
+                                    i + 30,
+                                    true,
+                                    (byte) 100,
+                                    (short) 200,
+                                    400L,
+                                    500.1f,
+                                    600.0d,
+                                    BinaryString.fromString(
+                                            i < 5 ? "update_string_" + i : "another_string_" + i),
+                                    Decimal.fromUnscaledLong(900, 5, 2),
+                                    Decimal.fromBigDecimal(new BigDecimal(1000), 20, 0),
+                                    TimestampLtz.fromEpochMillis(1698235273400L),
+                                    TimestampLtz.fromEpochMillis(1698235273400L, 7000),
+                                    TimestampNtz.fromMillis(1698235273501L),
+                                    TimestampNtz.fromMillis(1698235273501L, 8000),
+                                    new byte[] {5, 6, 7, 8},
+                                    BinaryString.fromString("test"),
+                                    0,
+                                    (long) 10 + i,
+                                    TimestampNtz.fromMillis(System.currentTimeMillis()))
+                            : row(
+                                    i + 30,
+                                    true,
+                                    (byte) 100,
+                                    (short) 200,
+                                    400L,
+                                    500.1f,
+                                    600.0d,
+                                    BinaryString.fromString(
+                                            i < 5 ? "update_string_" + i : "another_string_" + i),
+                                    Decimal.fromUnscaledLong(900, 5, 2),
+                                    Decimal.fromBigDecimal(new BigDecimal(1000), 20, 0),
+                                    TimestampLtz.fromEpochMillis(1698235273400L),
+                                    TimestampLtz.fromEpochMillis(1698235273400L, 7000),
+                                    TimestampNtz.fromMillis(1698235273501L),
+                                    TimestampNtz.fromMillis(1698235273501L, 8000),
+                                    new byte[] {5, 6, 7, 8},
+                                    0,
+                                    (long) 10 + i,
+                                    TimestampNtz.fromMillis(System.currentTimeMillis()));
+            expectedRows.add(convertToFlinkRow(row, fieldGetters));
+        }
+
+        List<Row> actualMergeRows =
+                mergeReadLakeRows(lakeSource, paimonSplits, fieldGetters, logRows);
+
+        assertThat(actualMergeRows).containsExactlyInAnyOrderElementsOf(expectedRows);
```

Comment:
Why do we need to test `actualMergeRows`? It looks a little of complex. 
For me, I think we only need to check the rows read by `SortedRecordReader` is ordered by the order return in method `SortedRecordReader#Comparator<InternalRow> order()`